src/oas/transformers/schema/__tests__/schema.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { translateSchemaObject } from '..';
+import { OASSchemaObject } from '../types';
+
+const translate = (schemaObject: OASSchemaObject) => translateSchemaObject(schemaObject, { pruneNotSupported: [] });
+
+describe('translateSchemaObject', () => {
+  it('should translate id', () => {
+    expect(
+      translate({
+        type: 'object',
+        properties: {
+          test: {
+            id: 'test',
+          },
+        },
+      }),
+    ).toStrictEqual({
+      $schema: 'http://json-schema.org/draft-07/schema#',
+      type: 'object',
+      properties: {
+        test: {
+          $id: 'test',
+        },
+      },
+    });
+  });
+
+  it('should translate exclusiveMinimum/exclusiveMaximum', () => {
+    expect(
+      translate({
+        type: 'integer',
+        minimum: 2,
+        maximum: 10,
+        exclusiveMinimum: true,
+        exclusiveMaximum: true,
+      }),
+    ).toStrictEqual({
+      $schema: 'http://json-schema.org/draft-07/schema#',
+      type: 'integer',
+      exclusiveMinimum: 2,
+      exclusiveMaximum: 10,
+    });
+  });
+
+  describe('OAS2 Schema Object', () => {
+    it('should translate x-nullable', () => {
+      expect(
+        translate({
+          type: 'string',
+          'x-nullable': true,
+        }),
+      ).toStrictEqual({
+        $schema: 'http://json-schema.org/draft-07/schema#',
+        type: ['string', 'null'],
+      });
+    });
+
+    it('should translate x-example', () => {
+      expect(
+        translate({
+          type: 'string',
+          'x-example': 'Cat',
+        }),
+      ).toStrictEqual({
+        $schema: 'http://json-schema.org/draft-07/schema#',
+        type: 'string',
+        'x-example': 'Cat', // kept for backwards compatibility, we may remove it in future (or maybe in this PR, let's figure it out!)
```

Comment:
How much work to get rid of it?

Reply:
I'm mostly concerned about JSE here.
It doesn't support arrayish examples, so we'd need to think of some UI for it.

OAS 3.1 says `example` is still valid, but `examples` is encouraged. 
`examples` is also more valid from JSON Schema's point of view.

Reply:
> AFAIK it doesn't support arrayish examples, so we'd need to think of some UI for it.

Reusing input we have for `enum` might be a good trade-off.
Are you good with it @marbemac?

Reply:
Hmm well the trick is that examples can occur anywhere right? Even on properties that are objects, which would not work well ui-wise w the simple tag input we use for enum 🤔.

Reply:
> Hmm well the trick is that examples can occur anywhere right? 

yeah, same as enum, and we don't support complex `enum`.

> Even on properties that are objects, which would not work well ui-wise w the simple tag input we use for enum.

Our current `example` is a plain text field, so from my point of view the change would be rather subtle

> which would not work well ui-wise w the simple tag input we use for enum thinking.

I agree, but as said, both example and enum fields are already limited to primitive values.

We can keep the existing `example` input if you wish, and simply do the conversion (`schema.examples = [example]`) later on, thoughts?

Reply:
If we are trying to keep the scope of the example > examples change to feature parity with what we currently have, then @P0lip is entirely right.

![image](https://user-images.githubusercontent.com/67381/115214688-ad9dcc00-a0fa-11eb-811f-d9f97cc0d51a.png)

We jut want to do this, and I believe the enum UI covers that nicely.

If we want to add "make an example for a whole object" then we can do that as a follow up feature, right?